code/modules/core_implant/cruciform/machinery/altar.dm

```diff
@@ -5,3 +5,40 @@
 	icon_state = "optable-idle"
 	y_offset = 10
 
+	var/list/acceptable_items = list(/obj/item/weapon/implant/core_implant/cruciform)
+	var/list/available_slots = list()
+
+/obj/machinery/optable/altar/New()
+	//bottom left
+	available_slots += list(list("offset" = list("x" = -8 , "y" = -3), "item" = null))
+	//bottom right
+	available_slots += list(list("offset" = list("x" = 8 , "y" = -3), "item" = null))
+
+/obj/machinery/optable/altar/attackby(obj/item/I, mob/user)
+	if(!istype(I) || !(I.type in acceptable_items))
+		return
+
+	item_cleanup()
+	var/list/slot = null
+	for(var/j = 1, j <= available_slots.len, j++)
+		slot = available_slots[j]
+		if (slot["item"] == null)
+			break
+		slot = null
+		
+	if(!slot)
+		to_chat(user, "<span class='notice'>There is no free space on \the [src] to place \the [I]!</span>")
+		return
```

Comment:
Im not sure if it's the right message because there will be plenty of space on the altar if there will be no occupant(at least visual) but it at least some feedback